### DISCUSSION TITLE
Producing Parquet format version 2?

### DISCUSSION THREAD

**Discussion by javafanboy** [jerolba/parquet-carpet#48] [Q&A]
I have used Carpet for a while as part of a personal project where I read and process the files using DuckDB, a tool I can really recommend for working with Parquet data by the way!

By chance I noticed that both pqrs and DuckDB reports the "version" of my produced Parquet files to be "1" .

As version "2" of the spec provides potential for more efficient encoding I would like to use it - is there anything special I need to do in Carpet to enable production of version 2 format or, version 2 is the default, some mistake I can make that would result in version 1?

**Comment by jerolba** [comment]
Hi @javafanboy

By default [Parquet Writer is configured with V1](https://github.com/apache/parquet-java/blob/master/parquet-column/src/main/java/org/apache/parquet/column/ParquetProperties.java#L53) and Carpet doesn't overwrite its default value. 

From my experience, in the world of Parquet there is no standard (even it have been defined) and different tools doesn't update to latest versions. In my job I've suffered to configure Carpet to generate files in versión 2 and break things in production because some python tools don't support it (Parquet V2 was released in December 2013!!). Then I decided to keep V1 as default version, and make it configurable.

To configure the writer with version 2 you need to do it via CarpetWriter builder:

```java
var writer = new CarpetWriter.Builder<>(outputStream, YourRecord.class)
    .withWriterVersion(WriterVersion.PARQUET_2_0)
    .build()
```

**Comment by javafanboy** [comment]
Thanks for the information and insightful opinion - will try it out and see if it works with my tools!
